### PR TITLE
Allow buttons to take icons and components that render icons

### PR DESCRIPTION
**Pull request by stredger**
## Description

Allow the button to render icons directly instead of having to pass a component in

### CODE REVIEW

File: packages/riipen-ui-docs/src/pages/components/button/Icons.jsx [hunk 1]
```diff
@@ -3,8 +3,6 @@ import React from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faChevronRight, faTrash } from "@fortawesome/free-solid-svg-icons";
 
-const icon = i => props => <FontAwesomeIcon icon={i} {...props} />;
-
```
Comment (stredger) [inline packages/riipen-ui-docs/src/pages/components/button/Icons.jsx:7]: allows us to get rid of this on web

File: packages/riipen-ui-docs/src/pages/components/button/Icons.jsx [hunk 2]
```diff
@@ -15,13 +13,17 @@ export default function Icons() {
 
   return (
     <div>
-      <Button color="default" iconStart={icon(faTrash)} variant="contained">
+      <Button
+        color="default"
+        iconStart={<FontAwesomeIcon icon={faTrash} />}
+        variant="contained"
+      >
```
Comment (stredger) [inline packages/riipen-ui-docs/src/pages/components/button/Icons.jsx:20]: and just use like this

File: packages/riipen-ui/src/components/Button.jsx [hunk 3]
```diff
@@ -463,12 +463,12 @@ Button.propTypes = {
...
    * Element placed after the children.
    */
-  iconEnd: PropTypes.elementType,
+  iconEnd: PropTypes.oneOfType([PropTypes.elementType, PropTypes.node]),
 
   /**
    * Element placed before the children.
    */
-  iconStart: PropTypes.elementType,
+  iconStart: PropTypes.oneOfType([PropTypes.elementType, PropTypes.node]),
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Button.jsx:471]: Nice! We should add tests to encompass this new functionality.

**Review by jordanell — CHANGES_REQUESTED** [review]

File: packages/riipen-ui/src/components/Button.jsx [hunk 4]
```diff
@@ -65,13 +65,13 @@ const Button = props => {
         <span className={clsx("label")}>
           {IconStart && (
             <span className={clsx("icon", "icon-start")}>
-              <IconStart />
+              {typeof IconStart === "function" ? <IconStart /> : IconStart}
```
Comment (daelynj) [inline packages/riipen-ui/src/components/Button.jsx:68]: Yeah definitely add tests for the cases when `IconStart` and End are functions and when they're not.

**Review by daelynj — APPROVED** [review]
Nice work!!

**Review by jordanell — APPROVED** [review]

**Pull request merged**